libcodechecker/analyze/analysis_manager.py

```diff
@@ -431,6 +464,26 @@ def check(check_data):
         return 1, skipped, reanalyzed, action.analyzer_type, None
 
 
+def create_actions_map(actions):
+    """
+    Create a dict for the build actions.
+    Key of the dict: (source_file, target).
+    """
+    result = dict()
+    for act in actions:
+        if act.source_count > 1:
+            LOG.error("Multiple sources for one build action: " +
+                      str(act.sources))
+        source = os.path.abspath(os.path.join(act.directory,
```

Comment:
Is it possible for `act.directory` to be relative? Usually, `os.path.abspath` is not the best solution most of the time since this is the path where CodeChecker was invoked and not the path where the compiler was invoked. (Which might be different).

Reply:
From here https://clang.llvm.org/docs/JSONCompilationDatabase.html:
`directory`: The working directory of the compilation. All paths specified in the `command` or `file` fields must be either absolute or relative to this directory.

If it was not an absolute path then all the other paths in `command` could not be relative paths, which is a contradiction with the definition of `directory`.
Consequently, it must be an absolute path.

Reply:
Ok, this means the `os.path.abspath` call there is redundant and should be removed.

Reply:
:+1: